go/vrf/ecvrf.go

```diff
@@ -169,3 +210,42 @@ func (p ECVRFParams) hashPoints(pm ...*big.Int) (c *big.Int) {
 	c = new(big.Int).SetBytes(cString[:n])
 	return c
 }
+
+// decodeProof
+//
+// Input: pi_string - VRF proof, octet string (ptLen+n+qLen octets)
+//
+// Output:
+//    Gx, Gy - Gamma - EC point
+//    c - integer between 0 and 2^(8n)-1
+//    s - integer between 0 and 2^(8qLen)-1
+//    or "INVALID"
+//
+// https://tools.ietf.org/html/draft-irtf-cfrg-vrf-06#section-5.4.4
+func (p ECVRFParams) decodeProof(pi []byte) (Gx, Gy, c, s *big.Int, err error) {
+	n := p.fieldLen / 2
+	if got, want := uint(len(pi)), p.ptLen+n+p.qLen; got != want {
+		return nil, nil, nil, nil, fmt.Errorf("len(pi): %v, want %v", got, want)
+	}
+
+	//    1.  let gamma_string = pi_string[0]...p_string[ptLen-1]
+	gStr := pi[:p.ptLen]
+	//    2.  let c_string = pi_string[ptLen]...pi_string[ptLen+n-1]
+	cStr := pi[p.ptLen : p.ptLen+n]
+	//    3.  let s_string =pi_string[ptLen+n]...pi_string[ptLen+n+qLen-1]
+	sStr := pi[p.ptLen+n : p.ptLen+n+p.qLen]
```

Comment:
It's probably more readable to just use `pi[p.ptLen+n : ]` here.

Reply:
Thanks